### ISSUE TITLE
checking is 'different' in .js files dependent on presence of //@ts-check comment

### ISSUE THREAD

**Issue by jrobinson01** [TypeStrong/atom-typescript#1471]
with comment:
![screen shot 2018-11-18 at 3 40 16 pm](https://user-images.githubusercontent.com/1584296/48677805-620d1f80-eb48-11e8-9a7e-83222e1abe8a.png)

no comment: 
![screen shot 2018-11-18 at 3 41 29 pm](https://user-images.githubusercontent.com/1584296/48677809-6fc2a500-eb48-11e8-8df2-4478af5276c8.png)

**Comment by lierdakil** [comment]
Well, this is not a bug, it's a feature. https://github.com/Microsoft/TypeScript/wiki/Type-Checking-JavaScript-Files

TL;DR: you need either this comment, or `checkJS: true` in relevant `tsconfig.json`.

**Comment by jrobinson01** [comment]
I'm not quite following. TS is doing "some" checking, but missing the particular case above. I'm not using a `tsconfig.json` file, but instead, a `jsconfig.json` as follows: 
```
{
  "compilerOptions": {
    "target": "es2015",
    "module": "es2015",
    "moduleResolution": "node",
    "lib": ["es2015", "dom", "es2016.array.include"]
  },
}
```

It seems like I'm running into this: https://github.com/Microsoft/TypeScript/wiki/Type-Checking-JavaScript-Files#object-literals-are-open-ended but changing to a typed object instead of enum also does not complain (unless I include the `// @ts-check` comment):
```
/** @type {{THING1:string}} */
const things = {
  THING1:'thing1',
};

/** @type {!string} */
var test = things.THING2;
```

**Comment by lierdakil** [comment]
Typescript doesn't check is files by default AFAIK...

пн, 19 нояб. 2018 г., 16:13 John Robinson <notifications@github.com>:

**Comment by jrobinson01** [comment]
I assume you mean "js" files. It seems that it can. It's generally working for me with the above jsconfig.json and using JSDoc annotations in my code, aside from the gotcha that it works better(?) if you also include the `// @ts-check` comment

**Comment by jrobinson01** [comment]
Alright, so it seems this is just a TS thing in general. I tried editing the same files with VS Code and get the same results.

**Comment by lierdakil** [comment]
https://github.com/Microsoft/TypeScript/wiki/Type-Checking-JavaScript-Files I linked above insists that TypeScript won't type-check js files if neither `checkJS: true` nor `// @ts-check` are in effect. So... yeah. Note that we can't, generally speaking, pass `--checkJS` to tsserver anyway, so the only reasonable way to go about this, as I'm seeing it, is to create `tsconfig.json` with `checkJS: true` and perhaps `include: 'src/**/*.js'` (or something to that effect). You'll most likely want to have `tsconfig.json` anyway at some point to configure typechecker strictness.

Note that `jsconfig.json` is a VSCode-specific feature. This is basically renamed `tsconfig.json` which, as far as I understand from skimming documentation, is ultimately fed to TypeScript compiler anyway. So...

**Comment by jrobinson01** [comment]
From what I’ve gathered, in order to check js files you either need checkJs:true in tsconfig.json or the comment. I’ve switched from jsconfig.json to tsconfig.json with checkJs and now all seems to be working as expected.
